docs/design/generics/details.md

```diff
@@ -2004,6 +2004,12 @@ interface NSpacePoint {
 }
 ```
 
+The pattern of an associated constant declaration must be a symbolic binding
+pattern, and unlike other `let` declarations, an associated constant declaration
+cannot have an initializer:
+
+_associated-constant-decl_ ::= `let` _identifier_ `:!` _expression_ `;`
```

Comment:
Is it worth saying the _expression_ is a type expression (meaning has an implicit cast to `type`)?

```suggestion
_associated-constant-decl_ ::= `let` _identifier_ `:!` _type-expression_ `;`
_associated-constant-decl_ ::= `default` `let` _identifier_ `:!` _type-expression_  = _init-expression_ `;`
```

Reply:
As in the other comment thread, no, I don't think so. This is a semantic distinction, not a syntactic one, so a grammar rule is the wrong place to express it.